### PR TITLE
refactor(studio): static tab kind registry and tab id codec (PR 1/9)

### PR DESCRIPTION
**Pull request by charislam**
## Summary

First PR in the tab/snippet decoupling stack. Purely additive: no existing call sites change behavior except the `EntityTypeIcon` rewire, which preserves its API exactly.

- **`state/tabs/kinds.ts`** — static `TAB_KINDS` descriptor table (`TabKind`, `TabSurface`, `surfaceOf`, `kindsOnSurface`, `isTabKind`). Leaf-safe: its only reference to the domain `Tab` type is a type-only import, so it stays importable at module scope (needed later for the valtio store and the persistence migration reader) without creating a runtime cycle.
- **`state/tabs/kinds.icons.tsx`** — per-kind icon leaf module (`ui` + `lucide-react` + `TabKind` type only), plus the preserved `LogsSnippetIcon`.
- **`state/tabs/tab-id.ts`** — `createTabId` / `parseTabId` / `toUrlSegment` / `parseUrlSegment(segment, surface)` codec. `parseUrlSegment` is surface-scoped: a bare segment only resolves to a kind when the surface has exactly one bare kind (true for `sql`). The table surface has five bare kinds (`r`/`v`/`m`/`f`/`p`) with no URL disambiguator between them, matching `/editor/[id]`, which learns kind from the fetched entity rather than the URL — so a bare table segment correctly resolves to nothing.
- **`components/ui/EntityTypeIcon.tsx`** — rewired to a thin wrapper delegating to `kinds.icons.tsx`, preserving its exact prop API (`type`, `size`, `strokeWidth`, `isActive`, `sqlSource`) for all existing consumers.
- **`state/tabs/tab-id.test.ts`** — codec round-trips per kind, bare-vs-prefixed URL segments, the `templates`/`examples`/`new` sentinels, and surface scoping (including the table-surface ambiguity above).

Full plan: `apps/studio/TABS_DECOUPLING_PLAN.md` (not included in this PR).

## Test plan

- [x] `pnpm typecheck`
- [x] `pnpm lint --filter=studio` (0 errors)
- [x] `pnpm --filter studio run lint:ratchet` (warning counts did not increase)
- [x] `pnpm test:studio` (full suite green, including 50 new/updated tests in `state/tabs/`)
- [x] `pnpm format` (no-op)

## Summary by CodeRabbit

* **New Features**
  * Added consistent icons and metadata for SQL, notebook, chat, table, view, and related tab types.
  * Added support for creating, parsing, and converting tab identifiers to URL segments.
  * Improved handling of tab types across SQL and table surfaces.

* **Bug Fixes**
  * Invalid, empty, or ambiguous tab identifiers and URL segments are now rejected.

* **Tests**
  * Added coverage for tab identifiers, URL conversion, supported tab types, and edge cases.

### CODE REVIEW

File: apps/studio/state/tabs/tab-id.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    const prefix = `${urlPrefix}-`
+    if (segment.startsWith(prefix)) {
+      const contentId = segment.slice(prefix.length)
+      if (contentId) return { kind, contentId }
+    }
+  }
+
+  return bareKind && !bareKindIsAmbiguous ? { kind: bareKind, contentId: segment } : null
```
[the remark this answers is not in the document]
Comment (charislam) [reply apps/studio/state/tabs/tab-id.ts]: sql here is the surface, not the kind

[the remark this answers is not in the document]
Comment (charislam) [reply apps/studio/state/tabs/tab-id.ts]: IDs are UUIDs in practice so this is impossible and there is no way to know the kind EXCEPT through this URL prefix.

**Review by joshenlim — APPROVED** [review]
Did a quick check on staging preview icons on SQL and Table Editor - LGTM!
